### DISCUSSION TITLE
Cleanup strategy for frequently changing containers

### DISCUSSION THREAD

**Discussion by aparcar** [containers/podman#20165] [General]
Hi, I'm running a services which starts multiple hundreds of containers a day and deletes them afterwards. Through this process a lot of dangling volumes and images appear, since the images themselves are updated daily. I came up with a trivial and likely terrible cleanup strategy and like to ask how more professional people handle this problem.

```
watch -n 10 podman system prune --volumes --force
```

This mechanism is running stable and fine, however I thought there would be something better. Ideally removing a container would remove its anonymous volumes and pulling a new container would automatically delete dangling images. Any advise?

**Comment by rhatdan** [comment]
Removing a container is supposed to remove it's anonymous volume.  There is a bug in quadlet, that I recently fixed, and you need to be careful if you are removing running containers via `podman rm --force`, there is a --volume flag that you should add to remove anonymous volumes.  I think this should default to true, but for now it is false.

**Reply by aparcar** [reply]
Thanks for your response, I'm already setting [`v=True`](https://github.com/openwrt/asu/blob/main/asu/common.py#L296) however this doesn't seem to do the trick. Any further ideas?

**Reply by rhatdan** [reply]
How are you removing them? What is your exact command?

@mheon Thoughts?

**Reply by mheon** [reply]
We really need to know how the containers are being created. Usually named volumes made without user intervention are going to be anonymous, and should be auto-removed when the container is removed, but there are definitely circumstances where this doesn't happen, and I'm honestly not sure whether this is the volumes not being marked as anonymous, or if the container removal is not removing anonymous volumes.

**Comment by rhatdan** [comment]
Also please give us the output of `podman info`
